x-pack/plugins/ml/public/application/trained_models/models_management/test_models/models/index_input.tsx

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useState, useMemo, useEffect, useCallback } from 'react';
+
+import useObservable from 'react-use/lib/useObservable';
+import { firstValueFrom } from 'rxjs';
+import { DataView } from '@kbn/data-views-plugin/common';
+import { EuiSpacer, EuiSelect, EuiFormRow, EuiAccordion, EuiCodeBlock } from '@elastic/eui';
+
+import { isPopulatedObject } from '@kbn/ml-is-populated-object';
+import { useMlKibana } from '../../../../contexts/kibana';
+import { RUNNING_STATE } from './inference_base';
+import type { InferrerType } from '.';
+
+interface Props {
+  inferrer: InferrerType;
+  data: ReturnType<typeof useIndexInput>;
+}
+
+export const InferenceInputFormIndexControls: FC<Props> = ({ inferrer, data }) => {
+  const {
+    dataViewListItems,
+    fieldNames,
+    selectedDataViewId,
+    setSelectedDataViewId,
+    selectedField,
+    setSelectedField,
+  } = data;
+
+  const runningState = useObservable(inferrer.runningState$);
+  const inputComponent = useMemo(() => inferrer.getInputComponent(), [inferrer]);
+
+  return (
+    <>
+      <EuiFormRow label="Index">
+        <EuiSelect
+          options={dataViewListItems}
+          value={selectedDataViewId}
+          onChange={(e) => setSelectedDataViewId(e.target.value)}
+          hasNoInitialSelection={true}
+          disabled={runningState === RUNNING_STATE.RUNNING}
+        />
+      </EuiFormRow>
+      <EuiSpacer size="m" />
+      <EuiFormRow label="Field">
```

Comment:
If you pick up an index without a `text` field it looks like it keeps a stale reference to the previous selection:

<img width="449" alt="image" src="https://user-images.githubusercontent.com/7405507/201658046-e483e307-c3b0-43a9-b75b-b42b85e8383f.png">

Can the Test button be disabled if there are no suitable fields, or could the list of indexes be filtered to only display indices with at least one text field?